.github/workflows/cd.yml

```diff
@@ -50,8 +26,67 @@ jobs:
     - name: Lint metadata file
       run: yamllint -d relaxed metadata.yaml
 
+  metadata:
+    needs: ["lint"]
+    runs-on: ubuntu-22.04
+    outputs:
+      latest_sha: ${{ steps.sha.outputs.LATEST_SHA }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Get latest sha from metadata file
+      id: sha
+      env:
+        METADATA_FILEPATH: 'metadata.yaml'
+      run: |
+        echo "LATEST_SHA=$(cat "${METADATA_FILEPATH}" | sed -n -e 's/^.*sha: \(.*\)$/\1/p' | head -n 1)" >> $GITHUB_OUTPUT
+
+
+  template:
+    needs: ["metadata"]
+    runs-on: ubuntu-22.04
+    outputs:
+      v_template: ${{ steps.template_version.outputs.TEMPLATE_VERSION }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ needs.metadata.outputs.latest_sha }}
```

Comment:
Should we specify a `path` here to have a second dedicated checkout and just merge the `template` and `metadata` jobs (possibly also with `version`)?

Reply:
Thank you very much for reviewing @jethron !
I really liked the idea about using the `path` so i merged `template` and `metadata` with `version`. Let me know what you think!